ENIGMAsystem/SHELL/Graphics_Systems/General/GStiles.cpp

```diff
@@ -43,66 +54,98 @@ namespace enigma
                     if (i==0){ prev_bkid = t.bckid; }
-                    draw_tile(index, t.bckid, t.bgx, t.bgy, t.width, t.height, t.roomX, t.roomY, t.xscale, t.yscale, t.color, t.alpha);
-
+                    draw_tile(vertexBuffer, t.bckid, t.bgx, t.bgy, t.width, t.height, t.roomX, t.roomY, t.xscale, t.yscale, t.color, t.alpha);
+                    vertex_count += 6;
                     if (prev_bkid != t.bckid || i == dit->second.tiles.size()-1){ //Texture switch has happened. Create new batch
                         get_background(bck2d,prev_bkid);
-                        drawing_depths[dit->second.tiles[0].depth].tilevector.push_back( vector< int >(3) );
-                        drawing_depths[dit->second.tiles[0].depth].tilevector.back()[0] = bck2d->texture;
-                        drawing_depths[dit->second.tiles[0].depth].tilevector.back()[1] = vert_start;
-                        drawing_depths[dit->second.tiles[0].depth].tilevector.back()[2] = vert_size;
-                        //printf("Texture id = %i and vertices to render = %i and start = %i\n", prev_bkid, vert_size, vert_start );
-                        vert_start += vert_size;
-                        vert_size = 0;
-                        //printf("Texture switch at i = %i and now texture = %i\n", i, prev_bkid );
+                        tile_layer_metadata[dit->second.tiles[0].depth].push_back( std::vector< int >(3) );
+                        tile_layer_metadata[dit->second.tiles[0].depth].back()[0] = bck2d->texture;
+                        tile_layer_metadata[dit->second.tiles[0].depth].back()[1] = vertex_start;
+                        tile_layer_metadata[dit->second.tiles[0].depth].back()[2] = vertex_count;
+
+                        vertex_start += vertex_count;
+                        vertex_count = 0;
+
                         prev_bkid = t.bckid;
                     }
-                    vert_size+=6;
-                    //printf("Tile = %i, tile x = %i and tile y = %i\n", i, t.bgx, t.bgy);
                 }
-                drawing_depths[dit->second.tiles[0].depth].tilevector.back()[2] += 6; //Add last quad
+                tile_layer_metadata[dit->second.tiles[0].depth].back()[2] += 6; //Add last quad
             }
         }
+
+        enigma_user::vertex_end(vertexBuffer);
     }
 
     void delete_tiles()
     {
         for (enigma::diter dit = drawing_depths.rbegin(); dit != drawing_depths.rend(); dit++){
             if (dit->second.tiles.size()){
-                drawing_depths[dit->second.tiles[0].depth].tilevector.clear();
-                enigma_user::d3d_model_destroy( drawing_depths[dit->second.tiles[0].depth].tilelist );
+                tile_layer_metadata[dit->second.tiles[0].depth].clear();
+                enigma_user::vertex_delete_buffer( tile_layer_buffers[dit->second.tiles[0].depth] );
             }
         }
     }
 
     void rebuild_tile_layer(int layer_depth)
     {
+        /*
```

Comment:
If this function is permanently obsolete, why comment it rather than delete it?

Reply:
Because I don't know if I should make it obsolete yet or not. I don't like the way it's designed and I had this discussion with Rusky already. Instead of rebuilding the tile model each time `tile_add` is called, I would like to defer building the tile model until the first `screen_redraw` where tiles have a "dirty" state. That way adjacent `tile_add` and `tile_delete` calls do not have as big of a performance hit.